### ISSUE TITLE
Trial runs on different machines

### ISSUE THREAD

**Issue by WerthPADOH** [PADOH-DHI/SAS-code#5]
I should have followed my own advice and started this as an Issue instead of through email.  Right now, the project needs to tested for universal working-ness on every type of setup and run-time environment used in the division.

Right now, this is the intended output (where `output_dir` is the directory you set in `config.sas`):

```
output_dir/
    ├ README.md
    ├ Functions/
    |   ├ README.md
    |   ├ functions.sas7bdat
    |   └ functions.sas7bndx
    └ Macros/
        ├ add_unpathed_members.sas
        └ clear_package_definitions.sas
```

If you encounter any errors in the code or problems with the output, please create an Issue with the bug label describing it.  That way, we can track what needs to be done and when it's done.

During the first run, Al encountered a problem (#4), which brings up what will likely be a common stumbling block: the division doesn't have a standard version of SAS.  For now, we'll just have to code around it.

For future reference, the original email:

> Hi guys,
> 
> I’ve pushed some changes to the SAS-code repository.  Most of these aim to make two things possible:
> 1.    Anyone can clone/download the project, do the minimum amount of setup, and run it successfully.
> 2.    The project can be run using the base SAS GUI, Enterprise Guide, or batch mode.
> 
> When you get the chance, please clone or download the project (green button on the right side of the repo’s main page).  If you have GitHub Desktop (which is pretty nice), you can try cloning it with that.
> 
> After you have a copy of the repo, open config-example.sas and change the value assigned to output_dir as whatever you want.  It has to be an existing directory, though.  After that, just run project_master.sas any way you’d like.  It would actually be beneficial to run it every way you can imagine.  The goal is to have the projects just work, no surprises.
> 
> Let me know how your testing goes.
> 
> Thanks

**Comment by rwright88** [comment]
config-example.sas needs to be renamed to config.sas, correct?

**Comment by WerthPADOH** [comment]
That's correct.  It has a separate name since config files have info that shouldn't be publicly available, so "config.sas" is ignored by git.  I figured an example config file in the repo would be the best way to show what's expected.

**Comment by rwright88** [comment]
I ran project_master.sas and got some errors so I uploaded the log.
[log.txt](https://github.com/PADOH-DHI/SAS-code/files/533917/log.txt)

**Comment by WerthPADOH** [comment]
What version of SAS do you have?

**Comment by rwright88** [comment]
NOTE: Copyright (c) 2002-2010 by SAS Institute Inc., Cary, NC, USA.
NOTE: SAS (r) Proprietary Software 9.3 (TS1M2)
      Licensed to CDC/ATSDR GRANTEE USE FOR SPONSORED PROGRAMS ONLY, Site 70084129.
NOTE: This session is executing on the W32_7PRO  platform.

NOTE: Enhanced analytical products:

SAS/STAT 12.1, SAS/ETS 12.1, SAS/OR 12.2, SAS/IML 12.1, SAS/QC 12.1

**Comment by WerthPADOH** [comment]
There is now a "sas9.3" branch, where the all code should work for SAS 9.3.  The bug-ridden sections have been removed from it.  Those bugs (#4 and #6) were part of content not required for the repo.  Use this branch for testing how universal the project can be, since the non-distribution-related code _should_ work.

The previous code is retained in the sas9.4 branch.  It's still available in the master branch's history, but this new branch has another purpose: housing future code requiring SAS 9.4.

**Comment by rwright88** [comment]
I cloned the sas9.3 branch and ran the program. I got a few more errors (log attached).

**Comment by rwright88** [comment]
I cloned the sas9.3 branch and ran the program. I got a few more errors.
[log.txt](https://github.com/PADOH-DHI/SAS-code/files/536161/log.txt)

**Comment by WerthPADOH** [comment]
Those errors are caused by my laziness.  I've been relying on SAS 9.4 to not throw a fit when passing non-temporary arrays to subroutines. 9.3 is stricter, and rightly so.  Hopefully fixed with efb16c6.

Also, did you clone the repo or save the zip file?  You should try working from a local repo and using the branches.  Switching branches will overwrite the files, which scared me at first, but it works really well.  You just have to commit changes to a branch before switching, and you can go back anytime.  Git even warns when you have uncommitted changes when you try to switch.

**Comment by rwright88** [comment]
I have a ticket in to get GitHub Desktop installed. Once it's installed I'll continue testing.
